src/ui/components/TestSuite/suspense/TestEventDetailsCache.ts

```diff
@@ -171,6 +185,64 @@ export const testEventDomNodeCache: ExternallyManagedCache<
   },
 });
 
+interface FindDOMNodeObjectIdForPersistentIdResult {
+  persistentId: string;
+  domNodeId: string;
+}
+
+const domNodeIdForPersistentIdCache: Cache<
+  [
+    replayClient: ReplayClientInterface,
+    point: TimeStampedPoint,
+    persistentIds: (string | number)[]
+  ],
+  { pauseId: string; domNodeIds: FindDOMNodeObjectIdForPersistentIdResult[] }
+> = createCache({
+  debugLabel: "DomNodeIdForPersistentIdCache",
+  getKey: ([replayClient, point, persistentIds]) => `${point}:${persistentIds.join()}`,
+  async load([replayClient, point, persistentIds]) {
+    if (findDOMNodeObjectIdForPersistentIdExpression === null) {
+      await lazyImportCypressExpression();
+    }
+
+    // The runtime expects these to be numbers, so we need to convert them.
+    const persistentIdsAsNumbers = persistentIds.map(id => Number(id));
+
+    const expression = `
+      const findDOMNodeObjectIdForPersistentId = ${findDOMNodeObjectIdForPersistentIdExpression};
+
+      const persistentIds = ${JSON.stringify(persistentIdsAsNumbers)};
+
+      const results = persistentIds.map(persistentId => {
+        const domNodeId = findDOMNodeObjectIdForPersistentId(persistentId)
+        return {
+          persistentId,
+          domNodeId,
+        }
+      })
+
+      const stringifiedResults = JSON.stringify(results);
+
+      // Implicit return
+      stringifiedResults;
+    `;
+
+    const pauseId = await pauseIdCache.readAsync(replayClient, point.point, point.time);
+    const result = await pauseEvaluationsCache.readAsync(replayClient, pauseId, null, expression);
+
+    const { returned } = result;
+    assert(returned?.value, "Expected a value to be returned from the eval.");
+    assert(typeof returned.value === "string", "Expected the value to be a string.");
```

Comment:
Oh I missed this the first time through but it's a nit so you can skip it if you don't think it's worthwhile.
```suggestion
    assert(typeof returned?.value === "string", "Expected a string to be returned from the eval.");
```